ModFrmHilD/Hecke.m

```diff
@@ -19,78 +19,55 @@ intrinsic HeckeOperator(f::ModFrmHilDElt, mm::RngOrdIdl : MaximalPrecision := fa
   K := CoefficientRing(f);
 
   coeffsTmmf := AssociativeArray();
-  prec := AssociativeArray();
   for bb in NarrowClassGroupReps(M) do
     coeffsTmmf[bb] := AssociativeArray();
-    prec[bb] := 0;
+    prec := Precision(Components(f)[bb]);
   end for;
+  new_prec := prec div Norm(mm);
 
   for bb in NarrowClassGroupReps(M) do
     bbp := NarrowClassGroupRepsToIdealDual(M)[bb];
     bbpinv := bbp^(-1);
 
-    for nu in ShintaniRepsUpToTrace(M, bb, Precision(M)) do //they come sorted
-      nn := nu*bbpinv;  // already call mm the ideal for the Hecke operator
+    for nu in FunDomainRepsUpToNorm(M, bb, new_prec) do //they come sorted
+      nn := nu*bbpinv;  // already call nn the ideal for the Hecke operator
       c := 0;
-      t := Integers()!Trace(nu);
-
 
       // loop over divisors
       // Formula 2.23 in Shimura - The Special Values
       // of the zeta functions associated with Hilbert Modular Forms
-      // If a coefficient in the sum is not defined we will set prec[bb] := Trace(nu) - 1;
       for aa in Divisors(ZF!!(nn + mm)) do
         if nn eq 0*ZF then
           //takes care if the coefficients for the zero ideal are different
           c +:= StrongMultiply(K, [* chi(aa), Norm(aa)^(k0 - 1), Coefficients(f)[NarrowClassRepresentative(M, bb*mm/aa^2)][ZF!0] *]);
         else
-          b, cf := IsCoefficientDefined(f, ZF!!(aa^(-2) * nn * mm));
-          if not b then
-            // stop looping through divisors if coefficient for at least one divisor
-            // is not defined (if trace (aa^(-2) * (nn*mm)) is greater than precision)
-            prec[bb] := t-1;
-            break; // breaks loop on aa
-          else
-            c +:= StrongMultiply(K, [* chi(aa),  Norm(aa)^(k0 - 1), cf *]);
+          // if nn * mm / aa^2 is an integral ideal
+          if nn * mm subset aa^2 then
```

Comment:
This will always holds if for `aa` in `Divisors(ZF!!(nn + mm))` right?

Reply:
Oh yes, thanks